cmd/ferretdb/ping.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// ping creates connection to FerretDB instance specified by the flags, and runs `ping` command against it.
+// The check is only executed if --setup-database flag is set.
+func ping() {
+	logger := setupLogger(cli.Log.Format, "")
+	checkFlags(logger)
+
+	l := logger.Sugar()
+
+	if cli.Setup.Database == "" {
+		l.Info("Setup database not specified - skipping ping.")
+		return
+	}
+
+	var urls []string
+
+	if cli.Listen.Addr != "" {
```

Comment:
What about `cli.Listen.TLS`?

Reply:
How do we want to implement it? We don't store CA key anywhere to sign temporary client certificate.

Reply:
Why do we need a temporary client certificate?.. Starting FerretDB with TLS requires passing all needed TLS files that we can use.
If your question is about "Extended Key Usage", I think we can safely ignore it, at least for now